### ISSUE TITLE
[Feature Request] Alternate template of save folder + AlbumID in tag & in folder name

### ISSUE THREAD

**Issue by viviancels** [DJDoubleD/QobuzDownloaderX-MOD#36]
# **🚀 Feature Request**

## **Is your feature request related to a problem? Please describe.**

*

---

## **Describe the solution you'd like**

*

---

## **Describe alternatives you've considered**

Hello.
1. Is it possible for the album to be saved according to the following template: **Track Artist - Album (Year)** or **Album Artist - Album (Year)**. If the album contains different artists, then you can specify VA instead of the performer or artist of first track. The fact is that sometimes different albums have the same album name and track name, in which case they add up to the same folder and there can be conflicts if they have the same track names
*
2. Very necessary option - AlbumID in tag or in folder name. Let take https://www.qobuz.com/us-en/album/another-dimention-dream-travel/rdsspixvxglqc - in this **rdsspixvxglqc** is album ID . So if added to it https://www.qobuz.com/us-en/album/ then we will get a link of album https://www.qobuz.com/us-en/album/rdsspixvxglqc. ("another-dimention-dream-travel" is no necessary part of the reference and it can be omitted). Example: **Dream Travel - Another Dimention (Original Mix) (2016) rdsspixvxglqc**
---

### **Additional context**

*

**Comment by gisturbance** [comment]
Issue https://github.com/DJDoubleD/QobuzDownloaderX-MOD/issues/24 for download template path.

**Comment by DJDoubleD** [comment]
As mentioned above, there are already some overlapping issue requesting customisable folder structures & filenames. Some of these requests conflict with eachother.
This isn't an easy feature to add currently, especially given the path lenght limitations on windows
.
Also, when downloading labels or artists, the album id is already added in square brackets as suffix to the album paths (exactly to avoid overriding different albums with the same name), and some people ask to remove that id from the path...
Will have to do some more thinking on how and if this can be resolved in the current app code...

**Comment by gisturbance** [comment]
@DJDoubleD path length should no longer be an [issue](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry#enable-long-paths-in-windows-10-version-1607-and-later).

**Comment by ElektroStudios** [comment]
+1

**Comment by viviancels** [comment]
Hello. For me, the topic is no longer relevant, since I move all downloaded labels and discographies automatically in mp3tag. But I want to add that it very often happens that an artist has two or more different releases with the same name, in this case, if you make a template for the name of the folder %artist% - %title of the release%, then there will be a danger of putting them all in one folder. Therefore, when automatically dragging and renaming downloaded releases to mp3tag, I either leave/add the album page ID like imf0jbp0gpdaa (Sometimes this ID is the same as UPC/Brarcode), and/or add a unique UPC/Barcode release number. (In particular cases, this number is not unique and different releases may have the same number, but this is already a lyric).

Some templates for mp3tag:
**Action name**: For label:
**Action**: Format value
**Field**: `_DIRECTORY`
**Format strings**: `..\..\'['%year%']' %albumartist% - %album% (%barcode%) $regexp(%_PATH%,'.* \[(.*)\].FLAC \(\d\dbit-.*?kHz\).*','$1')`
In this case, releases are moved to the downloaded label folder.

Be careful, windows explorer has path restrictions of 255 characters, which can be bypassed by editing the registry. And some files may not move. Moreover, Windows Explorer may not find files that have too long a name path and show that the folder in which they are located weighs 0 kilobytes! It is much more reliable to search for files in file managers such as total commander.

**Action name**: For discographie 1:
**Action**: Format value
**Field**: `_DIRECTORY`
Format strings: `..\'['%year%']' %albumartist% - %album% ($left(%year%,4))`
In this case, releases are moved to the 1 level up in folder structure.

**Action name**: For discographie 2:
**Action**: Format value
**Field**: `_DIRECTORY`
**Format strings**: `..\..\'['%year%']' %albumartist% - %album% ($left(%year%,4))`
In this case, releases are moved to the 2 level up in folder structure.

Also you can rename folders then:
**Action name**: Folder (1 [CanalogID] AlbumArtist - Album (Year)):
**Action**: Format value
**Field**: `_DIRECTORY`
**Format strings**: `'['%catalogID%']' %albumartist% - %album% ($left(%year%,4))`

**Action name**: Folder (1 [CanalogID] AlbumArtist - Album (Barcode)):
**Action**: Format value
**Field**: `_DIRECTORY`
**Format strings**: `'['%catalogID%']' %albumartist% - %album% (%Barcode%)`

**Action name**: Folder (2 [Year] AlbumArtist - Album (Barcode))
**Action**: Format value
**Field**: `_DIRECTORY`
**Format strings**: `'['%Year%']' %albumartist% - %album% (%Barcode%)`

**Action name**: Folder (2 [Year] AlbumArtist - Album (ALB_ID)):
**Action**: Format value
**Field**: `_DIRECTORY`
Format strings: `'['%Year%']' %albumartist% - %album% ($regexp(%AlbumID%,'.*\/(.*?)$','$1'))`

**Action name**: Folder (3 [Barcode] AlbumAlbum - Artist (Year)):
**Action**: Format value
**Field**: `_DIRECTORY`
**Format strings**: `'['%barcode%']' %albumartist% - %album% ($left(%year%,4))`

**Comment by orbittwz** [comment]
Dang 3 years almost from my initial comment, and maybe I can implement this soon after #116.
